server/src/main/java/org/elasticsearch/index/query/RegexpFlag.java

```diff
@@ -49,7 +46,7 @@ public enum RegexpFlag {
     /**
      * Enables complement expression of the form: {@code ~&lt;expression&gt;}
      */
-    COMPLEMENT(RegExp.COMPLEMENT),
+    COMPLEMENT(RegExp.DEPRECATED_COMPLEMENT),
```

Comment:
Is this something we need to deprecate/replace?

Reply:
Yes, I'll open a follow-up issue.  We have a choice here I think, between deprecating it and re-implementing it internally as a special case - see https://github.com/apache/lucene/pull/13739